app/src/androidTest/java/ch/epfl/cs311/wanderwave/model/BeaconConnectionTest.kt

```diff
@@ -0,0 +1,124 @@
+package ch.epfl.cs311.wanderwave.model
+
+import android.util.Log
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import ch.epfl.cs311.wanderwave.model.data.Track
+import ch.epfl.cs311.wanderwave.model.remote.BeaconConnection
+import io.mockk.junit4.MockKRule
+import kotlin.system.measureTimeMillis
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withTimeout
+import kotlinx.coroutines.withTimeoutOrNull
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+public class BeaconConnectionTest {
+
+  @get:Rule val mockkRule = MockKRule(this)
+  private lateinit var beaconConnection: BeaconConnection
+
+  @Before
+  fun setup() {
+    beaconConnection = BeaconConnection()
+  }
+
+  @Test
+  fun testAddAndGetItem() = runBlocking {
+    withTimeout(30000) { // Timeout after 5000 milliseconds (20 seconds)
```

Comment:
30000ms = 5000ms = 20s? ;)

Reply:
That's not my code, I pulled it from @KlaKalma 's branch to get the MapViewModel. Same applies for most of your other comments 😅

Reply:
Because it's the backend testing, I just want to be sure that it will not take too long. After 30000ms seemed to be a good middle ground. 
Sorry for the comments haha, it's tuning shit